### DISCUSSION TITLE
Adding support for hlw8032 energy monitor for 3 phase - request

### DISCUSSION THREAD

**Discussion by alonemars** [arendst/Tasmota#22161] [Support request]
Hi,

Is it possible add support for hlw8032 Energy monitor for 3 phase and report each phase through that functionality ? May be there is possibility to use 3 phase config for  hlw8032,I set the pins of D32, D35 and D34 of ESP32-Devkit V1 module to cse7766 Rx as the detection pin, but only the voltage and current of L1 line can be displayed in tasmota. What should I do to display the voltage and current and other information of phase 3?

**Comment by arendst** [comment]
?????

From the datasheet:

> Single-phase metering IC HLW8032

**Reply by alonemars** [reply]
I know HLW8032 is Single-phase metering IC, but is it possible to use tasmota to display three hlw8032 data?

**Reply by barbudor** [reply]
Do you know of a device that has 3x HLW8032 ?
If yes, probably possible but we need to see the diagram

**Reply by alonemars** [reply]
I am using ESP32 devkit1 and 3 hlw8032 ics to monitor the voltage and current information of my home. My setup is to connect the RX of the 3 ics to the pins D32,D5 and D34 of ESP32 devkit1 and set them to cse7766 RX respectively. But now only the information detected by one IC can be displayed. What am I doing wrong or tasmota can only display the detection information of one hlw8032?

**Reply by alonemars** [reply]
> Do you know of a device that has 3x HLW8032 ? If yes, probably possible but we need to see the diagram
This is the PCB I used

**Reply by barbudor** [reply]
As of today, the HLW8032 driver in Tasmota do not support more than 1 chip.
Not because it is not possible but because of the lack of existing devices that are built with 3 of them, this was not seen as usefull. 

You seem to have missed to attach the diagram 
Is that a commercial device or something DIY ?

**Reply by alonemars** [reply]
> As of today, the HLW8032 driver in Tasmota do not support more than 1 chip. Not because it is not possible but because of the lack of existing devices that are built with 3 of them, this was not seen as usefull.
> 
> You seem to have missed to attach the diagram Is that a commercial device or something DIY ?
This is a control board I DIY, the purpose is used to detect the voltage, current, and other information of the power supply, now the problem is that HLW8032 can not detect the 3 phase power supply at the same time :()
![pcb](https://github.com/user-attachments/assets/73203c38-9ee1-489b-bea2-c424edb6a768)
![diagram](https://github.com/user-attachments/assets/18b14d76-bf47-44b2-b2b4-b2b36ef4b7bd)
![pins](https://github.com/user-attachments/assets/78cbd21f-a3be-4202-8161-06d2c34f6041)
